lib/services/service-scaffold.ts

```diff
@@ -0,0 +1,217 @@
+	 */
+	protected queueData = (data: ServiceScaffoldServiceEvent) => {
+		if (this.verify(data)) {
+			super.queueEvent({
+				data,
+				workerMethod: this.handleEvent
+			});
+		} else {
+			this.logger.log(LogLevel.WARN, `Event failed verification.`);
+		}
+	}
+
+	/**
+	 * Emit a payload to the endpoint defined, resolving when done.
+	 * endpoint  Definition of the endpoint to emit to.
+	 * payload   Data to be delivered.
+	 */
+	protected abstract emitData(data: ServiceEmitContext): Promise<any>;
+
+	/**
+	 * Verify the event before enqueueing.  A naive approach could be to simply return true, but that must be explicit.
+	 * @param data  The data object that was enqueued.
+	 */
+	protected abstract verify(data: ServiceScaffoldServiceEvent): boolean
+
+	/**
+	 * Get a Worker object for the provided event, threaded by context.
+	 * @param event  Event as enqueued by the listener.
+	 * @returns      Worker for the context associated.
+	 */
+	protected getWorker = (event: ServiceScaffoldWorkerEvent): Worker<T> => {
+		// Attempt to retrieve an active worker for the context
+		const context = event.data.context;
+		const retrieved = this.workers.get(context);
+		if (retrieved) {
+			return retrieved;
+		}
+		// Create and store a worker for the context
+		const created = new Worker<T>(context, this.removeWorker);
+		this.workers.set(context, created);
+		return created;
+	}
+
+	/**
+	 * Create or retrieve the express app.
+	 * @returns  Express server app.
+	 */
+	protected get expressApp(): express.Express {
+		if (this.instanceExpressApp) {
+			return this.instanceExpressApp;
+		}
+		if (!ServiceScaffold.singletonExpressApp) {
+			// Either PROCBOT_PORT from environment or PORT from Heroku environment
+			const port = process.env.PROCBOT_PORT || process.env.PORT;
+			if (!port) {
+				throw new Error('No inbound port specified for express server.');
+			}
+			// Create and log an express instance
+			ServiceScaffold.singletonExpressApp = express();
+			ServiceScaffold.singletonExpressApp.use(bodyParser.json());
+			ServiceScaffold.singletonExpressApp.listen(port);
+			this.logger.log(LogLevel.INFO, `---> Started ServiceScaffold shared web server on port '${port}'.`);
+		}
+		return ServiceScaffold.singletonExpressApp;
+	}
+
+	/**
+	 * Retrieve the logger, here to write-protect it
+	 * @returns  Logger instance for this class
+	 */
+	protected get logger(): Logger {
+		return this.loggerInstance;
+	}
+
+	protected get eventsRegistered(): string[] {
+		return _.keys(this.eventListeners);
+	}
+
+	/**
+	 * Pass an event to registered listenerMethods.
+	 * @param data  Enqueued data from the listener.
+	 * @returns     Promise that resolves once the event is handled.
+	 */
+	protected handleEvent = (data: ServiceScaffoldServiceEvent): Promise<void> => {
+		// Retrieve and execute all the listener methods, squashing their responses
+		const listeners = this.eventListeners[data.type] || [];
+		return Promise.map(listeners, (listener) => {
```

Comment:
This'll fail on the first error that occurs, which means that the events won't get sent to any of the other listeners. You probably want a catch on the call to the `listenerMethod` to ensure that it doesn't (see `github.ts` for an example).

Reply:
Note to self: Done.